### PR TITLE
Add future and throwing to convertToPublic

### PR DESCRIPTION
**Pull request by BrettRToomey**

### CODE REVIEW

File: Sources/JWTKeychain/JWTKeychainUser/JWTKeychainUser+JWTKeychainUserType.swift [hunk 1]
```diff
@@ -45,7 +45,7 @@ extension JWTKeychainUser: JWTKeychainUserType {
         }
     }
 
-    public func convertToPublic() -> Public {
-        return .init(email: email, name: name)
+    public func convertToPublic(on db: DatabaseConnectable) throws -> Future<Public> {
```
Comment (siemensikkema) [inline Sources/JWTKeychain/JWTKeychainUser/JWTKeychainUser+JWTKeychainUserType.swift]: Is DatabaseConnectable also a Container? Is this the best abstraction level?

Comment (BrettRToomey) [reply Sources/JWTKeychain/JWTKeychainUser/JWTKeychainUser+JWTKeychainUserType.swift]: It was a basic worker. I'm not sure where to get both. Should I just take in the request?

Comment (siemensikkema) [reply Sources/JWTKeychain/JWTKeychainUser/JWTKeychainUser+JWTKeychainUserType.swift]: Yeah, That's the most versatile one. A bit restrictive maybe but I don't see when we shouldn't use request (outside of tests, but even there it's relatively easy).

Comment (BrettRToomey) [reply Sources/JWTKeychain/JWTKeychainUser/JWTKeychainUser+JWTKeychainUserType.swift]: Sounds good.

Comment (BrettRToomey) [reply Sources/JWTKeychain/JWTKeychainUser/JWTKeychainUser+JWTKeychainUserType.swift]: @siemensikkema done

**Review by siemensikkema — APPROVED** [review]

**Pull request merged**